winch/filetests/filetests/x64/store/f64.wat

```diff
@@ -0,0 +1,30 @@
+;;! target = "x86_64"
+
+(module
+  (memory (data "\00\00\00\00\00\00\f4\7f"))
+  (func (export "f64.store") (f64.store (i32.const 0) (f64.const nan:0x4000000000000)))
+)
+
+;;      	 55                   	push	rbp
+;;      	 4889e5               	mov	rbp, rsp
+;;      	 4883ec08             	sub	rsp, 8
+;;      	 4d8b5e08             	mov	r11, qword ptr [r14 + 8]
+;;      	 4d8b1b               	mov	r11, qword ptr [r11]
+;;      	 4939e3               	cmp	r11, rsp
+;;      	 0f8722000000         	ja	0x3a
+;;   18:	 4c893424             	mov	qword ptr [rsp], r14
+;;      	 f20f10051c000000     	movsd	xmm0, qword ptr [rip + 0x1c]
+;;      	 b800000000           	mov	eax, 0
+;;      	 498b4e50             	mov	rcx, qword ptr [r14 + 0x50]
+;;      	 4801c1               	add	rcx, rax
```

Comment:
This pattern shows up in a few places, would it be easy to special case the `0` offset and avoid the `add`?

Reply:
Yeah, a bit unfortunate, there are a couple of patterns like this throughout Winch. I was hoping that we would be able to get rid of them once we start exploring the optimizations that we could apply ; right now this would require introducing at least a new abstraction at the `CodeGenContext` to enable peeking at constants at certain points of the code generation. This is not too difficult I think, but before getting into any optimizations for the generated code, I was hoping to make all the possible improvements to the compilation times to have a good sense of what our baseline is.